src/Confluent.Kafka/Consumer.cs

```diff
@@ -414,6 +414,28 @@ public void Unassign()
             => consumer.Unassign();
 
         /// <summary>
+        ///     Store offsets for one or more partitions.
+        ///     
+        ///     The offset will be committed (written) to the offset store according
+        ///     to `auto.commit.interval.ms` or manual offset-less commit().
+        /// </summary>
+        /// <remarks>
+        ///     `enable.auto.offset.store` must be set to "false" when using this API.
```

Comment:
@mhowlett , I am about to add the validation. There are two validations that must be performed.
1. if enbale.auto.commit not true, then throw an exception. If you look at my most recent commit, I have updated the remarks saying this setting must be set to true.
2. As you requested if enable.auto.offset.store is true, also throw an exception. 

Don't you think these validations must be performed inside Librdkafka library rather than here? If you still strongly feel, please confirm and I will add them here.

Reply:
I suggest not adding validations to language bindings, but rather to librdkafka, if needed.

It is perfectly fine to have auto commits disabled and use offsets_store(), the stored offsets will be used on your next argument-less commit() call.

For the validation of enable.auto.offset.store being disabled, I think that has questionable implementation complexity / payoff ratio. We already document the fact as a remark on the offset_store API itself.

Reply:
@edenhill , thank you for the info about offsets_store() can be invoked when auto commit is disabled. I will remove the line from remarks that says 'enable.auto.commit' must be true.

Reply:
Done, removed 'enable.auto.commit' must be true from remarks.

Reply:
> For the validation of enable.auto.offset.store being disabled, I think that has questionable implementation complexity / payoff ratio

I disagree on this. We should fail fast when we know the user has configured something incorrectly. Kafka is complex, people don't read docs, and they configure it incorrectly all the time and get frustrated. We should take any opportunity we have to push people towards the right thing. Many small details like this add up to ease of use.

However, I agree that this check should be in librdkafka, not `Confluent.Kafka`, so this PR needs no additions.